src/sentry/utils/snuba.py

```diff
@@ -54,6 +54,8 @@
 # doesn't include new lines,
 QUOTED_LITERAL_RE = re.compile(r"^'[\s\S]*'$")
 
+MEASUREMENTS_KEY_RE = re.compile(r"^measurements\.([a-zA-Z0-9-_.]+)$")
```

Comment:
The [relay pattern doesn't allow numbers](https://github.com/getsentry/relay/blob/master/relay-general/src/protocol/measurements.rs#L56), should we accept them here?

Reply:
Hm... that seems to have changed since the original spec. I'll remove the numbers from this pattern.

Reply:
The error message in relay also mentions numbers. Perhaps @dashed knows more about what changed?

Reply:
Numbers are allowed. This is right 👍

Reply:
I'll submit a fix by tomorrow. But it shouldn't be an issue considering web vital names don't contain numbers.

Reply:
Hm... Should I remove the upper case letters from the pattern? Relay normalizes the measurement name to only contain lower case letters. Or perhaps automatically convert them to lower case somewhere.

Reply:
Uppercase is still legal. But we should lowercase it before hitting snuba

Reply:
To clarify, measurement names can have uppercase letters in the event payload before it's sent to Sentry. But Relay will lower case the letters before it's passed to Snuba for indexing.

Reply:
Addressed in b44923d

Reply:
Relay PR for allowing numbers: https://github.com/getsentry/relay/pull/785